GDJS/GDJS/Extensions/Builtin/CommonInstructionsExtension.cpp

```diff
@@ -408,6 +409,87 @@ CommonInstructionsExtension::CommonInstructionsExtension() {
+            event.GetConditions().empty()
+                ? "true"
+                : codeGenerator.GenerateBooleanFullName(
+                      "condition" +
+                          gd::String::From(event.GetConditions().size() - 1) +
+                          "IsTrue",
+                      context) +
+                      ".val";
+
+        // Prepare object declaration and sub events
+        gd::String subevents =
+            codeGenerator.GenerateEventsListCode(event.GetSubEvents(), context);
+        gd::String objectDeclaration =
+            codeGenerator.GenerateObjectsDeclarationCode(context) + "\n";
+
+        // Write final code
+        gd::String iteratorReferenceVariableName =
+            "iteratorReference" +
+            gd::String::From(context.GetContextDepth());
+        gd::String structureChildVariableName =
+            "structureChild" +
+            gd::String::From(context.GetContextDepth());
+        gd::String structureReferenceVariableName =
+            "structureReference" +
+            gd::String::From(context.GetContextDepth());
+
+        outputCode +=
+            "const $ITERATOR_VARIABLE = runtimeScene.getVariables().get(\"$CHILD_VARIABLE\");\n"
+            "const $STRUCTURE_REFERENCE = runtimeScene.getVariables().get(\"$STRUCTURE_VARIABLE\");\n"
+            "for(\n"
+            "    let $STRUCTURE_CHILD_VARIABLE in \n"
+            "    $STRUCTURE_REFERENCE.getAllChildren()\n"
+            ") {\n"
+            "    $STRUCTURE_CHILD_VARIABLE = $STRUCTURE_REFERENCE.getChild($STRUCTURE_CHILD_VARIABLE)\n"
+            "    if($STRUCTURE_CHILD_VARIABLE.isNumber()) {\n"
+            "        $ITERATOR_VARIABLE.setNumber($STRUCTURE_CHILD_VARIABLE.getAsNumber())\n"
+            "    } else if ($STRUCTURE_CHILD_VARIABLE.isStructure()) {\n"
+            "        // Structures are passed by reference like JS objects\n"
+            "        $ITERATOR_VARIABLE._children = $STRUCTURE_CHILD_VARIABLE.getAllChildren()\n"
+            "    } else {\n"
+            "        $ITERATOR_VARIABLE.setString($STRUCTURE_CHILD_VARIABLE.getAsString())\n"
+            "    }\n";
+        outputCode += objectDeclaration;
+        outputCode += conditionsCode;
+        outputCode += "if (" + ifPredicat + ")\n";
+        outputCode += "{\n";
+        outputCode += actionsCode;
+        if (event.HasSubEvents()) {
+          outputCode += "\n{ //Subevents: \n";
+          outputCode += subevents;
+          outputCode += "} //Subevents end.\n";
+        }
+        outputCode += "}\n";
+        outputCode += "}\n";
+
+        return outputCode.FindAndReplace("$STRUCTURE_CHILD_VARIABLE", structureChildVariableName)
+            .FindAndReplace("$ITERATOR_VARIABLE", iteratorReferenceVariableName)
+            .FindAndReplace("$STRUCTURE_VARIABLE", event.GetStructure())
+            .FindAndReplace("$STRUCTURE_REFERENCE", structureReferenceVariableName)
+            .FindAndReplace("$CHILD_VARIABLE", event.GetVariable());
```

Comment:
$CHILD_VARIABLE => $ITERATOR_VARIABLE_NAME

```suggestion
            .FindAndReplace("$ITERATOR_VARIABLE_NAME", event.GetIteratorVariableName());
```